### ISSUE TITLE
Loading scrim always hidden if list is empty

### ISSUE THREAD

**Issue by nwhittaker** [Esri/calcite-design-system#10120]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Given a `<calcite-list loading />` element with a non-zero height and no children, the loading scrim is not visible.

<img width="494" alt="Screenshot 2024-08-20 at 4 00 21 PM" src="https://github.com/user-attachments/assets/4766303d-1766-4863-b1cf-1e9d9a7d5f78">

### Expected Behavior

The loading scrim is fully or partially visible depending on how much height the consumer has given the list. In other words, if the list is a scrolling container, the scrim appears centered in the container regardless of how many rows there may be, if any.

<img width="494" alt="Screenshot 2024-08-20 at 4 00 13 PM" src="https://github.com/user-attachments/assets/b98ca166-2ece-4b2c-a9d1-7d1098ec4197">

### Reproduction Sample

https://codepen.io/nwhittaker-esri/pen/JjQLXJg

### Reproduction Steps

1. Visit the sample and see the loading scrim is not visible.

### Reproduction Version

2.11.1

### Priority impact

impact - p2 - want for an upcoming milestone

### Impact

The desire is to be able to use the `loading` state when initially populating an expensive list.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Field Apps

**Comment by driskull** [comment]
The `list` component is currently displayed as `block` by default. If we were to change it to `flex` that would be a visual breaking change.

**Comment by nwhittaker** [comment]
>The list component is currently displayed as block by default. If we were to change it to flex that would be a visual breaking change.

Agreed. If the issue can be fixed without changing the list's display, that'd be ideal.

**Comment by geospatialem** [comment]
cc https://github.com/Esri/calcite-design-system/issues/5162

**Comment by ashetland** [comment]
@geospatialem We're removing the design label from this. We agree the long term story here is improvements related to #5162, but that effort shouldn't prevent moving forward with @driskull's [fix](https://github.com/Esri/calcite-design-system/issues/10120#issuecomment-2307356544) in the meantime.

**Comment by geospatialem** [comment]
Work towards a min-height for the fix.

**Comment by isaacbraun** [comment]
Verified on `3.3.0-next.51` via https://codepen.io/isaacbraun/pen/xbwgrZd.

**Comment by jcfranco** [comment]
@isaacbraun @geospatialem Are we no longer considering this a potential breaking change as @driskull [suggested](https://github.com/Esri/calcite-design-system/issues/10120#issuecomment-2307356544)?

**Comment by jcfranco** [comment]
Awesome. Thanks for confirming!
